src/AutoRest.CSharp/AutoRest/Plugins/CSharpProj.cs

```diff
@@ -31,7 +33,21 @@ internal class CSharpProj : IPlugin
         public async Task<bool> Execute(IPluginCommunication autoRest)
         {
             var ns = await autoRest.GetValue<string>("namespace");
-            await autoRest.WriteFile($"{Configuration.ProjectRelativeDirectory}{ns}.csproj", _csProjContent, "source-file-csharp");
+
+            string codeModelFileName = (await autoRest.ListInputs()).FirstOrDefault();
+            if (string.IsNullOrEmpty(codeModelFileName))
+                throw new Exception("Generator did not receive the code model file.");
+
+            Task<CodeModel> codeModelTask = Task.Run(async () =>
+            {
+                var codeModelYaml = await autoRest.ReadFile(codeModelFileName);
+                return CodeModelSerialization.DeserializeCodeModel(codeModelYaml);
+            });
+
+            var codeModel = await codeModelTask;
+            var defaultLibraryName = codeModel.Language.Default.Name;
+
+            await autoRest.WriteFile($"{Configuration.ProjectRelativeDirectory}{ns ?? defaultLibraryName}.csproj", _csProjContent, "source-file-csharp");
```

Comment:
we have too many places where we implement this fallback logic

Reply:
We don't have access to the [BuildContext.cs](https://github.com/Azure/autorest.csharp/blob/56c302316386a361a96565f920069651416cc306/src/AutoRest.CSharp/Output/Models/Types/BuildContext.cs#L27) here. That's why added this logic here.

Reply:
We can create it. We don't need the source info part.